frontend/pages/queries/QueryPage/components/QueryResults/QueryResultsTableConfig.tsx

```diff
@@ -47,28 +47,25 @@ const _unshiftHostname = (headers: IDataColumn[]) => {
   return newHeaders;
 };
 
-const generateResultsTableHeaders = (results: unknown[]): Column[] => {
-  // Table headers are derived from the shape of the first result.
-  // Note: It is possible that results may vary from the shape of the first result.
-  // For example, different versions of osquery may have new columns in a table
-  // However, this is believed to be a very unlikely scenario and there have been
-  // no reported issues.
-  const shape = results[0];
-  const keys =
-    shape && typeof shape === "object" && isPlainObject(shape)
-      ? Object.keys(shape)
-      : [];
+const generateResultsTableHeaders = (results: any[]): Column[] => {
+  const keys = Array.from(
+    results.reduce(
+      (s, o) => Object.keys(o).reduce((t, k) => t.add(k), s),
+      new Set() // Set prevents listing duplicate headers
+    )
+  );
```

Comment:
I think it would be more clear to either put a comment or pull out the code here into a named function to help understand what this code is doing.

Reply:
Done!